Source/JavaScriptCore/runtime/JSString.cpp

```diff
@@ -106,37 +106,41 @@ void JSString::visitChildrenImpl(JSCell* cell, Visitor& visitor)
     JSString* thisObject = asString(cell);
     ASSERT_GC_OBJECT_INHERITS(thisObject, info());
     Base::visitChildren(thisObject, visitor);
-    
     uintptr_t pointer = thisObject->fiberConcurrently();
-    if (pointer & isRopeInPointer) {
-        if (pointer & JSRopeString::isSubstringInPointer) {
-            visitor.appendUnbarriered(static_cast<JSRopeString*>(thisObject)->fiber1());
-            return;
-        }
-        for (unsigned index = 0; index < JSRopeString::s_maxInternalRopeLength; ++index) {
-            JSString* fiber = nullptr;
-            switch (index) {
-            case 0:
-                fiber = std::bit_cast<JSString*>(pointer & JSRopeString::stringMask);
-                break;
-            case 1:
-                fiber = static_cast<JSRopeString*>(thisObject)->fiber1();
-                break;
-            case 2:
-                fiber = static_cast<JSRopeString*>(thisObject)->fiber2();
-                break;
-            default:
-                ASSERT_NOT_REACHED();
-                return;
-            }
-            if (!fiber)
-                break;
+
+    // This JSString is not JSRopeString. Thus we report held string's cost.
+    if (cell->perCellBit()) {
```

Comment:
Can we add an `bool JSString::isStringImplOwner() const { return perCellBit(); }` and use that instead of the `perCellBit()` directly? I think it would make this code easier to follow in the future. Might also be worth a comment about the new ownership model at the top of `JSRopeString` since the comment is out of date now. e.g. "To make concurrent collector work correctly, we must initialize 2nd and 3rd fibers at JSRopeString creation and we must not modify these part later."